zerver/tests/test_auth_backends.py

```diff
@@ -1627,6 +1610,42 @@ class GitHubAuthBackendTest(SocialAuthBase):
     AUTH_FINISH_URL = "/complete/github/"
     CONFIG_ERROR_URL = "/config-error/github"
 
+    def social_auth_test_finish(self, result: HttpResponse,
+                                account_data_dict: Dict[str, str],
+                                headers: Dict[str, Any]={},
+                                expect_choose_email_screen: bool=False,
```

Comment:
I think some of the `headers` / `expect_choose_email_screen` refactoring refactoring is counterproductive, since it obscures what's happening and squashies together unrelated objects.  I undid that before merging.